docs/snippets/common/storybook-interactions-play-function.js.mdx

```diff
@@ -0,0 +1,27 @@
+```js
+// MyForm.stories.js
+import { expect } from '@storybook/jest';
+import { userEvent, waitFor, within } from '@storybook/testing-library';
+import { MyForm } from './MyForm'
+
+export default {
+  title: 'MyForm',
+  component: MyForm,
+  argTypes: {
+    onSubmit: { action: true },
+  },
+};
+
+const Template = (args) => <MyForm {...args} />;
+
+const Submitted = Template.bind({});
+Submitted.play = async ({ canvasElement }) => {
+  const canvas = within(canvasElement);
+
+  await userEvent.type(canvas.getByTestId('email'), 'hi@example.com');
+  await userEvent.type(canvas.getByTestId('password'), 'supersecret');
+  await userEvent.click(canvas.getByRole('button'));
+
+  await waitFor(() => expect(args.onSubmit).toHaveBeenCalled());
```

Comment:
this args is being used but not defined in the play function signature

Reply:
also, shouldn't `expect` always be awaited?

Reply:
`waitFor` takes an async function, aka a function that returns a Promise. `expect` returns a Promise, so there's no need to wrap it in an async/await block, we can just return the result of expect directly. Of course this only works if you only want to `waitFor` a single thing.